types/compilation/compilation.interfaces.ts

```diff
@@ -121,3 +122,28 @@ export type ToolResult = {
     stdout: ResultLine[];
     artifact?: Artifact;
 };
+
+export type CompilationInfo = {
+    mtime: any;
```

Comment:
`lib/handle/compiler.js` in line 155 sets the `modificationTime` equal to `Stat.mtime` which is a `Date`. This also gives us a proper type for base-compiler's `mtime`, but that probably will need to be `Date | null` given that we only set it when calling `initialize`

Reply:
Or TIL, https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#strict-class-initialization is useful for this cases